imports/plugins/core/catalog/server/methods/catalog.js

```diff
@@ -0,0 +1,241 @@
+    if (variant.inventoryManagement) {
+      return Catalog.getVariantQuantity(variant) <= 0;
+    }
+    return false;
+  });
+}
+
+/**
+ * @method isLowQuantity
+ * @summary If at least one of the variants is less than the threshold, then function returns `true`
+ * @memberof Catalog
+ * @param {Array} variants - array of child variants
+ * @return {boolean} low quantity or not
+ */
+export function isLowQuantity(variants) {
+  return variants.some((variant) => {
+    const quantity = Catalog.getVariantQuantity(variant);
+    // we need to keep an eye on `inventoryPolicy` too and qty > 0
+    if (variant.inventoryManagement && variant.inventoryPolicy && quantity) {
+      return quantity <= variant.lowInventoryWarningThreshold;
+    }
+    return false;
+  });
+}
+
+/**
+ * @method isBackorder
+ * @summary Is products variants is still available to be ordered after summary variants quantity is zero
+ * @memberof Catalog
+ * @param {Array} variants - array with variant objects
+ * @return {boolean} is backorder allowed or not for a product
+ */
+export function isBackorder(variants) {
+  return variants.every((variant) => !variant.inventoryPolicy && variant.inventoryManagement &&
+    variant.inventoryQuantity === 0);
+}
+
+/**
+ * @method publishProductToCatalog
+ * @summary Publish a product to the Catalog
+ * @memberof Catalog
+ * @param {string} productId - A string product id
+ * @return {boolean} true on successful publish, false if publish was unsuccessful
+ */
+export async function publishProductToCatalog(productId) {
+  check(productId, String);
+
+  // Find the product by id
+  let product = Products.findOne({
+    $or: [
+      { _id: productId },
+      { ancestors: { $in: [productId] } }
+    ]
+  });
+
+  // Stop of a product could not be found
+  if (!product) {
+    Logger.info("Cannot publish product to catalog");
+    return false;
+  }
+
+  // If the product has ancestors, then find to top product document
+  if (Array.isArray(product.ancestors) && product.ancestors.length) {
+    product = Products.findOne({
+      _id: product.ancestors[0]
+    });
+  }
+
+  // Get variants of the product
+  const variants = Products.find({
+    ancestors: {
+      $in: [productId]
+    }
+  }).fetch();
+
+  // Get Media for the product
+  const mediaArray = await Media.find({
+    "metadata.productId": productId,
+    "metadata.toGrid": 1,
+    "metadata.workflow": { $nin: ["archived", "unpublished"] }
+  }, {
+    sort: { "metadata.priority": 1, "uploadedAt": 1 }
+  });
+
+  // Denormalize media
+  const productMedia = mediaArray.map((media) => ({
+    metadata: media.metadata,
+    thumbnail: `${media.url({ store: "thumbnail" })}`,
+    small: `${media.url({ store: "small" })}`,
+    medium: `${media.url({ store: "medium" })}`,
+    large: `${media.url({ store: "large" })}`,
+    image: `${media.url({ store: "image" })}`
+  }));
+
+  // Denormalize product fields
+  product.media = productMedia;
+  product.type = "product-simple";
```

Comment:
Is this a new `type`? Why don't we use `simple` or `variant` like in other places?

Reply:
Yes. For the catalog, products of type `simple` will be `product-simple` as it's more descriptive and because other types of data may live in the Catalog, not just products.